phc/services/tools.py

```diff
@@ -0,0 +1,281 @@
+"""A Python Module for Files"""
+
+import os
+import math
+import backoff
+from phc.base_client import BaseClient
+from phc import ApiResponse
+from urllib.parse import urlencode
+from urllib.request import urlretrieve
+from phc.errors import ApiError
+
+
+class Tools(BaseClient):
+    """Provides acccess to PHC tools registry
+
+    Parameters
+    ----------
+    session: phc.Session
+        The PHC session
+    run_async: bool
+        True to return promises, False to return results (default is False)
+    timeout: int
+        Operation timeout (default is 30)
+    trust_env: bool
+        Get proxies information from HTTP_PROXY / HTTPS_PROXY environment variables if the parameter is True (False by default)
+    """
+
+    def create(
+        self,
+        name: str,
+        description: str,
+        access: str,
+        version: str,
+        tool_class: str,
+        source: str,
+        labels: str = None,
+    ) -> ApiResponse:
+        """Create a tool.
+
+        Parameters
+        ----------
+        name: str
+            The name to give to the tool
+        description: str
+            A description of the tool
+        access: str
+            The access level given to the tool [PRIVATE, ACCOUNT, PHC, PUBLIC]
+        version: str
+            The initial version of the tool
+        tool_class: str
+            The class of the tool [Workflow, Notebook]
+        source: str
+            The path of the tool to upload
+        labels: str, optional
+            A comma delimted list of labels to apply to the tool, i.e. 'bam,samtools'
```

Comment:
Can we make this a proper Python list and then join it with the comma separator? It would make the input significantly clearer.

Reply:
Sure, if that's more clear.  The reason I did it this why is to keep it consistent with the `cli` so that the two appear somewhat the same, not sure that's worth sustaining though.